test/Util/TestUtilImage.cc

```diff
@@ -0,0 +1,296 @@
+/* This file is part of the CARTA Image Viewer: https://github.com/CARTAvis/carta-backend
+   Copyright 2018- Academia Sinica Institute of Astronomy and Astrophysics (ASIAA),
+   Associated Universities, Inc. (AUI) and the Inter-University Institute for Data Intensive Astronomy (IDIA)
+   SPDX-License-Identifier: GPL-3.0-or-later
+*/
+
+#include <casacore/casa/Exceptions/Error.h>
+
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include "Util/Image.h"
+
+class ImageUtilTest : public ::testing::Test {};
+
+TEST_F(ImageUtilTest, DefaultAxisRangeConstructor) {
+    AxisRange range;
+    EXPECT_EQ(range.from, 0);
+    EXPECT_EQ(range.to, ALL_Z);
+}
+
+TEST_F(ImageUtilTest, SingleValueAxisRangeConstructor) {
+    AxisRange range(5);
+    EXPECT_EQ(range.from, 5);
+    EXPECT_EQ(range.to, 5);
+}
+
+TEST_F(ImageUtilTest, AxisRangeConstructor) {
+    AxisRange range(2, 8);
+    EXPECT_EQ(range.from, 2);
+    EXPECT_EQ(range.to, 8);
+}
+
+TEST_F(ImageUtilTest, RangeEqualityOperator) {
+    AxisRange range1(3, 7);
+    AxisRange range2(3, 7);
+    AxisRange range3(4, 7);
+    AxisRange range4(3, 8);
+
+    EXPECT_TRUE(range1 == range2);
+    EXPECT_FALSE(range1 == range3);
+    EXPECT_FALSE(range1 == range4);
+}
+
+TEST_F(ImageUtilTest, RangeInequalityOperator) {
+    AxisRange range1(3, 7);
+    AxisRange range2(4, 7);
+
+    EXPECT_TRUE(range1 != range2);
+    EXPECT_FALSE(range1 != range1);
+}
+
+TEST_F(ImageUtilTest, IsInAxisRange) {
+    AxisRange range(10, 20);
+    EXPECT_TRUE(range.is_in_range(15));
+    EXPECT_TRUE(range.is_in_range(10));
+    EXPECT_TRUE(range.is_in_range(20));
+    EXPECT_FALSE(range.is_in_range(9));
+    EXPECT_FALSE(range.is_in_range(21));
+}
+
+TEST_F(ImageUtilTest, NegativeAxisRange) {
+    AxisRange range(-10, -5);
+    EXPECT_TRUE(range.is_in_range(-7));
+    EXPECT_TRUE(range.is_in_range(-10));
+    EXPECT_TRUE(range.is_in_range(-5));
+    EXPECT_FALSE(range.is_in_range(-11));
+    EXPECT_FALSE(range.is_in_range(-4));
+}
```

Comment:
We don't expect to use AxisRange like this -- real values should always be positive; we only use negative values for placeholders (like ALL_Z or CURRENT_Z).